### PR TITLE
Makefile: Use 'git diff' to show gofmt changes

### PR DESCRIPTION
**Pull request by wking**
This makes fixing errors easier.  Before this commit, errors looked [like][1]:

```console
$ make gofmt
!!! 'gofmt -s' needs to be run on the following files:
./lib/config.go
make: *** [gofmt] Error 1
```

But that's not very helpful when your local `gofmt` thinks the file is fine.  With this commit, errors will look like:

```console
$ make gofmt
diff --git a/lib/config.go b/lib/config.go
index 1acca8c7..6a63b2b0 100644
--- a/lib/config.go
+++ b/lib/config.go
@@ -2,7 +2,7 @@ package lib

 import (
         "bytes"
-"io/ioutil"
+       "io/ioutil"

        "github.com/BurntSushi/toml"
        "github.com/kubernetes-incubator/cri-o/oci"
make: *** [Makefile:67: gofmt] Error 1
```

(or whatever, I just stuffed in a formatting error for demonstration purposes).

[1]: https://travis-ci.org/kubernetes-incubator/cri-o/jobs/331949394#L1075

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,7 +64,8 @@ lint: .gopathok
 	@./.tool/lint
 
 gofmt:
-	@./hack/verify-gofmt.sh
+	find . -name '*.go' ! -path './vendor/*' -exec gofmt -s -w {} \+
```
Comment (mrunalp) [inline Makefile]: Isn't this actually running gofmt on the code? If so, it won't display any issues in the code in a PR.

Comment (wking) [reply Makefile]: > Isn't this actually running gofmt on the code?

Yes.

> If so, it won't display any issues in the code in a PR.

Right.  But the next line's `git diff ...` will show them (and return non-zero) if anything changed.

Comment (mrunalp) [reply Makefile]: Doesn't quite work for me locally:
```
[mrunalp@localhost cri-o]$ git diff
diff --git a/server/container_create.go b/server/container_create.go
index 506bcf79..492d8eea 100644
--- a/server/container_create.go
+++ b/server/container_create.go
@@ -295,7 +295,7 @@ func addDevices(sb *sandbox.Sandbox, containerConfig *pb.ContainerConfig, specge
                                UID:   &hostDevice.Uid,
                                GID:   &hostDevice.Gid,
                        }
-                       if hostDevice.Major == 0 && hostDevice.Minor == 0 {
+                               if hostDevice.Major == 0 && hostDevice.Minor == 0 {
                                // Invalid device, most likely a symbolic link, skip it.
                                continue
                        }
[mrunalp@localhost cri-o]$ make gofmt
find . -name '*.go' ! -path './vendor/*' -exec gofmt -s -w {} \+
git diff --exit-code
[mrunalp@localhost cri-o]$ echo $?
0
[mrunalp@localhost cri-o]$ git diff
[mrunalp@localhost cri-o]$
```

Comment (mrunalp) [reply Makefile]: nvm, I see the point. It would have shown the bad changes if present in the bad commit.

**Comment by TomSweeneyRedHat** [comment]
LGTM, neat little change!

**Comment by rhatdan** [comment]
LGTM
/test all

**Comment by runcom** [comment]
this doesn't need e2e :confused:

**Pull request merged**